blockchain/blockdao/blockdao.go

```diff
@@ -573,64 +533,52 @@ func (dao *blockDAO) putBlock(blk *block.Block) error {
 	if h != hash.ZeroHash256 && err == nil {
 		return errors.Errorf("block %d already exist", blkHeight)
 	}
-
-	serHeader, err := blk.Header.Serialize()
-	if err != nil {
-		return errors.Wrap(err, "failed to serialize block header")
-	}
-	serBody, err := blk.Body.Serialize()
+	serBlk, err := blk.Serialize()
 	if err != nil {
-		return errors.Wrap(err, "failed to serialize block body")
-	}
-	serFooter, err := blk.Footer.Serialize()
-	if err != nil {
-		return errors.Wrap(err, "failed to serialize block footer")
+		return errors.Wrap(err, "failed to serialize block")
 	}
 	if dao.compressBlock {
 		timer := dao.timerFactory.NewTimer("compress_header")
-		serHeader, err = compress.Compress(serHeader)
+		serBlk, err = compress.Compress(serBlk)
 		timer.End()
 		if err != nil {
-			return errors.Wrapf(err, "error when compressing a block header")
-		}
-		timer = dao.timerFactory.NewTimer("compress_body")
-		serBody, err = compress.Compress(serBody)
-		timer.End()
-		if err != nil {
-			return errors.Wrapf(err, "error when compressing a block body")
-		}
-		timer = dao.timerFactory.NewTimer("compress_footer")
-		serFooter, err = compress.Compress(serFooter)
-		timer.End()
-		if err != nil {
-			return errors.Wrapf(err, "error when compressing a block footer")
+			return errors.Wrapf(err, "error when compressing a block")
 		}
 	}
-	batchForBlock := db.NewBatch()
-	hash := blk.HashBlock()
-	batchForBlock.Put(blockHeaderNS, hash[:], serHeader, "failed to put block header")
-	batchForBlock.Put(blockBodyNS, hash[:], serBody, "failed to put block body")
-	batchForBlock.Put(blockFooterNS, hash[:], serFooter, "failed to put block footer")
 	kv, _, err := dao.getTopDB(blkHeight)
 	if err != nil {
 		return err
 	}
+	htf, err := kv.CreateRangeIndexNX([]byte(blockDataNS), make([]byte, 8))
```

Comment:
should use CountingIndex, instead of range index

Reply:
I am not sure it can be added sequentially,CountingIndex dont have insert method,add method only add value